# Graded assignment 1

Iris Troeijen 


# Packages

We need the package `tidyverse`

In [20]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [21]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2        V3        V4        V5        V6        V7        
1 0.12184005 0.5023009 0.6281145 0.3628500 0.2878019 0.3332426 0.56159010
2 0.03741853 0.9860663 0.8523578 0.9142504 0.3847193 0.4104716 0.05517048
  V8        V9        V10        V11       V12       V13       V14      
1 0.4796252 0.7512985 0.82332616 0.7994653 0.6394688 0.2827766 0.3700889
2 0.3989097 0.6854330 0.06851253 0.4931964 0.0494151 0.4096830 0.5368931
  V15       V16        V17       V18       V19       V20        
1 0.2761682 0.03259931 0.9434325 0.5940146 0.7419795 0.009793917
2 0.2006211 0.16067460 0.7730413 0.7868981 0.5404655 0.772513974

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [23]:
df2 <- df1 %>% 
    gather(Key, Value, V1:V20)
head(df2)

Key Value     
1 V1  0.12184005
2 V1  0.03741853
3 V2  0.50230089
4 V2  0.98606626
5 V3  0.62811446
6 V3  0.85235778

## Assignment 2

We will generate some data:

In [24]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [25]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [26]:
df5 <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook